ui/jobs/buff_tracker.ts

```diff
@@ -46,19 +90,19 @@ export class Buff {
         secondsUntilShow, this.cooldownSortKeyBase, 'grey', '', 0.5, addReadyCallback);
   }
 
-  addReady(source) {
+  addReady(source: string): void {
     if (this.ready[source]) {
       // Unexpected use of the same cooldown by the same name.
-      this.ready[source].removeCallback();
+      this.ready[source]?.removeCallback();
     }
 
     // TODO: could consider looking at the party list to make initials unique?
     let txt = '';
     const initials = source.split(' ');
     if (initials.length === 2)
-      txt = initials[0][0] + initials[1][0];
+      txt = `${initials[0]?.charAt(0) ?? '?'}${initials[1]?.charAt(0) ?? '?'}`;
     else
-      txt = initials[0];
+      txt = initials[0] ?? '?';
```

Comment:
Is there something that can make this more readable?

Maybe `txt = initials.map((str) => str?.charAt(0) ?? '?').join('');`